### PR TITLE
feat(datagrid): implement skeleton with the loading state

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
implement skeleton with the loading state

**What is the chosen solution to this problem?**

apply also the change in the PieChartButton.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

* component: PieChartButton
* PR: [feat(datagrid): implement skeleton with the loading state. change inProgress by loading in the PieChartButton](https://github.com/Talend/ui/pull/1127)
* Changes:

| Before     | After   |
| ---------- | ------- |
| inProgress | loading |

### CODE REVIEW

File: packages/components/stories/Skeleton.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
 
-const icons = {
-	'talend-locked': talendIcons['talend-locked'],
-};
-
 stories
-	.addDecorator(story => <div className="col-lg-offset-2 col-lg-8">{story()}</div>)
+	.addDecorator(story => (
+		<div className="col-lg-offset-2 col-lg-8">
+			<IconProvider />
```
Comment (romainseb) [inline packages/components/stories/Skeleton.js]: All icons are stored in the snapshot this way.
With selected icons passed for the test, this was not the case.

Comment (jare-talend) [reply packages/components/stories/Skeleton.js]: why do you need to test IconProvider. it's not necesseray.

Comment (romainseb) [reply packages/components/stories/Skeleton.js]: It's here to make the icons work, it's not test as itself 🤔

File: packages/components/src/PieChartButton/PieChartButton.test.js [hunk 2]
```diff
@@ -13,11 +13,90 @@ import {
...
+			const wrapper = shallow(
+				<PieChartButton label="my label" display="small" model={pieChartData} onClick={onClick} />,
+			);
+
+			wrapper
+				.find('Button')
+				.at(0)
+				.simulate('click', event);
+
+			expect(onClick).toHaveBeenCalledWith(event, {
```
Comment (jsomsanith) [inline packages/components/src/PieChartButton/PieChartButton.test.js]: Please split in 2 tests : render and action

File: packages/components/src/PieChartButton/PieChartButton.test.js [hunk 3]
```diff
@@ -13,11 +13,90 @@ import {
...
+				color: 'slate-gray',
+				percentage: 4,
+			},
+			{
+				color: 'silver-chalice',
+				percentage: 3,
+			},
+		];
+
+		it('shoud render a PieChartButton', () => {
```
Comment (jsomsanith) [inline packages/components/src/PieChartButton/PieChartButton.test.js]: Typo `should`

File: packages/components/src/PieChartButton/PieChartButton.test.js [hunk 4]
```diff
@@ -13,11 +13,90 @@ import {
...
+				action: {
+					label: 'my label',
+				},
+				model: pieChartData,
+			});
+
+			expect(wrapper.getElement()).toMatchSnapshot();
+		});
+
+		it('shoud render a PieChartButton with an overlay', () => {
```
Comment (jsomsanith) [inline packages/components/src/PieChartButton/PieChartButton.test.js]: Typo `should`

File: packages/components/src/PieChartButton/PieChartButton.test.js [hunk 5]
```diff
@@ -13,11 +13,90 @@ import {
...
+					onClick={onClick}
+				/>,
+			);
+
+			wrapper
+				.find('Button')
+				.at(0)
+				.simulate('click', event);
+
+			expect(onClick).not.toHaveBeenCalledWith(event, {
```
Comment (jsomsanith) [inline packages/components/src/PieChartButton/PieChartButton.test.js]: Same here,
* title is `should render a PieChartButton with an overlay`
* you test a click, that should not trigger onClick
Not really clear in my head

File: packages/datagrid/src/components/DataGrid/DataGrid.scss [hunk 6]
```diff
@@ -1,8 +1,23 @@
...
 .td-grid {
 	color: $black;
 	height: 100%;
 	width: 100%;
 
+	&-in-progress {
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		height: 100%;
```
Comment (jsomsanith) [inline packages/datagrid/src/components/DataGrid/DataGrid.scss]: height is already defined to this value, please remove this line

File: packages/datagrid/src/components/DataGrid/DataGrid.test.js [hunk 7]
```diff
@@ -190,6 +190,12 @@ describe('#DataGrid', () => {
 		expect(getPinnedColumnDefsFn).toHaveBeenCalledWith(sample);
 		expect(getRowDataFn).toHaveBeenCalledWith(sample);
 	});
+
+	it('should render one Skeleton is loading', () => {
+		const wrapper = shallow(<DataGrid loading getComponent={getComponent} />);
+		expect(wrapper.find('Skeleton').length).toBe(1);
```
Comment (jsomsanith) [inline packages/datagrid/src/components/DataGrid/DataGrid.test.js]: This is tested implicitely with snapshot, remove this line please

File: packages/datagrid/stories/datagrid.container.js [hunk 8]
```diff
@@ -176,6 +187,16 @@ storiesOf('Container Datagrid')
 		),
 		options,
 	)
+	.addWithCMF(
+		'in Progress',
```
Comment (jsomsanith) [inline packages/datagrid/stories/datagrid.container.js]: Rename to `loading` please

File: packages/datagrid/stories/datagrid.component.js [hunk 9]
```diff
@@ -5,15 +5,22 @@ import { IconsProvider } from '@talend/react-components';
...
+			<DataGrid
+				data={sample}
+				onFocusedCell={event => console.log(event)}
+				onFocusedColumn={event => console.log(event)}
+				onRowsSelected={event => console.log(event)}
+				rowSelection="multiple"
+			/>
+		</div>
+	))
+	.add('in progress', () => (
```
Comment (jsomsanith) [inline packages/datagrid/stories/datagrid.component.js]: Rename to `loading` please

**Pull request merged**